### PR TITLE
Pylint cleanups for gen/db/*.py

### PR DESCRIPTION
**Pull request by cdhorn**

### CODE REVIEW

File: gramps/gen/db/generic.py [hunk 1]
```diff
@@ -311,27 +325,15 @@ def __next__(self):
     def __exit__(self, *args, **kwargs):
         pass
 
-    def iter(self):
```
Comment (Nick-Hall) [inline gramps/gen/db/generic.py:314]: I don't think that we should remove the `iter`, `first` and `next` methods.  They form part of the public API and third-party code may be using them.  If we want to remove these in the future then they should be deprecated first.

**Review by Nick-Hall — CHANGES_REQUESTED** [review]
The only problem I can see with this is the removal of a few `Cursor` methods.

I generally don't mind the abbreviations "trans" for "transaction" or "dir" for "directory", but in the database code maybe the full form is clearer.  Keep the changes  you have made to these.

There are also a couple a pylint comments.

File: gramps/gen/db/generic.py (hunk 1 above)
Comment (cdhorn) [reply gramps/gen/db/generic.py]: Restored them.

**Comment by cdhorn** [comment]
Removed the pylint hints as well, think this is ready.

**Review by Nick-Hall — APPROVED** [review]

**Pull request merged**